addons/knowledge/static/src/js/knowledge_renderers.js

```diff
@@ -321,13 +321,46 @@ const KnowledgeArticleFormRenderer = FormRenderer.extend(KnowledgeTreePanelMixin
             items: 'li',
             listType: 'ul',
             toleranceElement: '> div',
-            forcePlaceholderSize: true,
             opacity: 0.6,
             placeholder: 'bg-info',
             tolerance: 'pointer',
             helper: 'clone',
             cursor: 'grabbing',
             cancel: '.readonly',
+            /**
+             * Prevent a non-root shared article from becoming one, because the
+             * access rights cannot be inferred in that case.
+             *
+             * @param {jQuery} placeholder jQuery element which represents the
+             *                 destination position (may be the placeholder or
+             *                 the nestedSortableItem if it already moved)
+             * @param {jQuery} placeholderParent undefined, required by the
+             *                 library (mjs.nestedSortable) function signature
+             * @param {jQuery} currentItem jQuery nestedSortableItem element
+             * @returns {boolean}
+             */
+            isAllowed: (placeholder, placeholderParent, currentItem) => {
+                return (
+                    // destination is under another article
+                    placeholder[0].parentElement.closest('.o_article') ||
+                    // destination is not within the shared section
+                    !placeholder[0].closest('section[data-section="shared"]') ||
+                    // starting position was a root of the shared section
+                    currentItem.data('nestedSortableItem').offsetParent[0].parentElement.getAttribute('data-section') === "shared"
+                );
+            },
+            /**
+             * Prevent the display of a placeholder in the root shared section
+             * if the current item cannot be dragged there.
+             *
+             * @param {Event} event
+             * @param {Object} ui
+             */
+            start: (event, ui) => {
+                if (ui.item.data('nestedSortableItem').offsetParent[0].parentElement.getAttribute('data-section') !== 'shared') {
```

Comment:
That's a pretty long selector right there. Some would tell you to keep it like that for the sake of conciseness. Some would tell you to introduce an intermediate variable for the sake of clarity. I think I belong to the second category of people :smile:

 I suggest to introduce an intermediate variable to store the selected element.
```js
const section = ui.item.data('nestedSortableItem').offsetParent[0].parentElement;
if (section.getAttribute('data-section') {
    // ...
}
```

(same for the `isAllowed` function)